File: lab1_2_12/3.py
import sys

def main():
    try:
        line = input("Enter row, col of mat 1 & mat 2 : ").split()
        if not line: return
        r1, c1, r2, c2 = map(int, line)
        
        if c1 != c2 or r1 != r2:
            print("Addition not possible")
            return

        print("Enter mat 1")
        mat1 = []
        for _ in range(r1):
            mat1.append(list(map(int, input().split())))
        
        print("Enter mat 2")
        mat2 = []
        for _ in range(r1):
            mat2.append(list(map(int, input().split())))
            
        mat3 = [[0 for _ in range(c1)] for _ in range(r1)]
        for i in range(r1):
            for j in range(c1):
                mat3[i][j] = mat1[i][j] + mat2[i][j]

        print("Sum matrix")
        for row in mat3:
            print(*(row))
            
    except (ValueError, EOFError, IndexError):
        pass

if __name__ == "__main__":
    main()

File: lab1_2_12/3.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int r1, c1, r2, c2;
    cout << "Enter row, col of mat 1 & mat 2 : ";
    cin >> r1 >> c1 >> r2 >> c2;
    if(c1 != c2 && r1 != r2)
    {
        cout << "Addition not possible\n";
        return 0;
    }
    cout << "Enter mat 1\n";
    vector<vector<int>> mat1(r1, vector<int>(c1,0));
    for(int i=0;i<r1;i++)
        for(int j=0;j<c1;j++)
            cin >> mat1[i][j];
    
    cout << "Enter mat 2\n";
    vector<vector<int>> mat2(r1, vector<int>(c1,0));
    for(int i=0;i<r1;i++)
        for(int j=0;j<c1;j++)
            cin >> mat2[i][j];           
    
    vector<vector<int>> mat3(r1, vector<int>(c1,0));
    for(int i = 0;i<r1;i++)
        for(int j = 0;j<c1;j++)
            mat3[i][j] = mat1[i][j] + mat2[i][j];

    cout << "Sum matrix\n";

    for(int i =0;i<r1;i++)
    {
        for(int j = 0;j<c1;j++)
            cout << mat3[i][j] << " ";
        cout << "\n";
    }
}